### DISCUSSION TITLE
Integration Tests

### DISCUSSION THREAD

**Discussion by akoenig** [vercel/turborepo#3986] [Help] [answered]
I came across an interesting use case where I'm currently struggling to wrap my head around it. What would be the suggested approach to implement integration tests? The workflow looks like:

1. Boot containers with infrastructure services (in our case a database)
2. Execute test suite
3. Teardown

This is how we implemented the `pipeline` so far:

```json
    "prisma:generate:schema": {
      "inputs": ["prisma/**/*.{ts,sql}"],
      "outputs": ["prisma/schema.prisma"]
    },
    "prisma:generate:client": {
      "cache": false,
      "dependsOn": ["prisma:generate:schema"]
    },
    "test:before": {
      "dependsOn": ["prisma:generate:client"],
      "cache": false
    },
    "test": {
      "dependsOn": ["test:before"],
      "inputs": ["src/**"],
      "outputs": [".jest/**"]
    },
```

Now, it turned out that the `test:before` gets executed although the `inputs` in `test` did not change. Wondering if this is something we messed up or if this is actually a bug.

Any hint into the right direction is highly appreciated 🙂

**Comment by mehulkar** [comment, marked as answer]
In your task graph, it seems like `test` eventually depends on `prisma:generate:schema`, which _does_ have `inputs` and `outputs` defined. Although `test:before` will run every time (especially because it has `cache:false` configured, it seems that unless you make a change to `prisma/**/*.{ts,sql}`, nothing should actually be happening before `test`.
